### PR TITLE
Symfony 3 compatibility first commit

### PR DESCRIPTION
**Pull request by enachesilviu**
Installed dependencies with composer. Tested on PHP 7.

PHPUnit 5.1.0 by Sebastian Bergmann and contributors.

................................................................. 65 / 68 ( 95%)
...                                                               68 / 68 (100%)

Time: 237 ms, Memory: 16.00Mb

OK (68 tests, 252 assertions)

Also did a basic fanout queue manual test with numbers from 0 to 999, published them using a symfony command (on a newly created Symfony 3 project) and sent them to screen using a consumer.

This is the first time I do a public PR, please tell me if anything goes wrong. Thank you.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,19 +9,19 @@
...
-        "symfony/dependency-injection" : "~2.3",
-        "symfony/event-dispatcher"     : "~2.3",
-        "symfony/config"               : "~2.3",
-        "symfony/yaml"                 : "~2.3",
-        "symfony/console"              : "~2.3",
+        "symfony/dependency-injection" : "~2.3|~3.0",
+        "symfony/event-dispatcher"     : "~2.3|~3.0",
+        "symfony/config"               : "~2.3|~3.0",
+        "symfony/yaml"                 : "~2.3|~3.0",
+        "symfony/console"              : "~2.3|~3.0",
```
Comment (videlalvaro) [inline composer.json]: so to be clear this let's people use symfony from version 2.3 an forward, or version 3.0 and forward?

**Comment by videlalvaro** [comment]
can you add something to the `.travis.yml` file so test are run form Symfony 3 as well?

File: composer.json (hunk 1 above)
Comment (enachesilviu) [inline composer.json]: Yes, this is what I mean.
Also changed the description of the package a bit to let people know it works on both 2 and 3 series. If you want me to revert this allow me to know.

**Comment by enachesilviu** [comment]
I'm not accustomed to Travis but the file seemed rather intuitive so I made my additions to it. I hope it's ok, otherwise I'm going to need help on this one.

**Comment by enachesilviu** [comment]
As I said before I tested this on PHP 7. I got it from remi's repo installed on a CentOS 7 machine:

http://rpms.famillecollet.com/enterprise/7/php70/x86_64/repoview/php.html

**Comment by enachesilviu** [comment]
I don't know about Travis but I just tested this on PHP 5.3.3 on a CentOS 6.7 machine. After the last commit it works:

[root@localhost rabbit]# phpunit
PHPUnit 4.8.19 by Sebastian Bergmann and contributors.

................................................................. 65 / 68 ( 95%)
...

Time: 927 ms, Memory: 36.75Mb

OK (68 tests, 252 assertions)
[root@localhost rabbit]# php --version
PHP 5.3.3 (cli) (built: Jul  9 2015 17:39:00)
Copyright (c) 1997-2010 The PHP Group
Zend Engine v2.3.0, Copyright (c) 1998-2010 Zend Technologies
[root@localhost rabbit]# cat /etc/*release
CentOS release 6.7 (Final)
CentOS release 6.7 (Final)
CentOS release 6.7 (Final)

**Comment by videlalvaro** [comment]
There's an error, the PHP version required here is wrong for Symfony 3 https://travis-ci.org/videlalvaro/RabbitMqBundle/jobs/95007127

**Comment by enachesilviu** [comment]
Travis does this:  export SYMFONY_VERSION="3.0.*" . It adds symfony/framework-bundle series 3 in composer.json require key. It will never work with old php.

Instead we can add in composer.json require this: "symfony/framework-bundle"     : "~2.3 || ~3.0" and it works with the rest of requirements. I tried this, here's the composer.lock file:

http://pastebin.com/sx24zGAm

**Comment by enachesilviu** [comment]
Did last commit (373cddd) for the purpose of testing with travis. I know travis.yml looks weird now.

File: composer.json [hunk 2]
```diff
@@ -8,20 +8,21 @@
...
     }],
     "require": {
-        "php"                          : ">=5.3.0",
-        "symfony/dependency-injection" : "~2.3",
-        "symfony/event-dispatcher"     : "~2.3",
-        "symfony/config"               : "~2.3",
-        "symfony/yaml"                 : "~2.3",
-        "symfony/console"              : "~2.3",
-        "videlalvaro/php-amqplib"      : "~2.6"
+        "php"                          : ">=5.3.0 || >=5.5.9",
```
Comment (Tobion) [inline composer.json]: this change makes no sense as >=5.5.9 is useless

Comment (enachesilviu) [inline composer.json]: Ok, reverted this.

The last problem I encountered was the wrong version of PHPUnit for PHP 5.5 series. For this specific series PHPUnit is always the wrong version. If we could fix this and the tests give us a thumb up maybe we can proceed with having this bundle ready for Symfony 3 series.

**Comment by alinalexandru** [comment]
@enachesilviu  I'll try to help you with the travis job. Some tests are broken, are not working bellow SF 2.6

**Comment by ruudk** [comment]
I figured it out in this PR: https://github.com/videlalvaro/RabbitMqBundle/pull/313

For PHP 5.5 I PHPUnit 4.8 :)

**Comment by ruudk** [comment]
#313 is merged, this one can be closed.

**Comment by enachesilviu** [comment]
PR 313 merged, does the same thing.

**Pull request closed without merging**